plugins/modules/grafana_folder.py

```diff
@@ -281,24 +305,32 @@ def get_version(self):
             return {"major": int(major), "minor": int(minor), "rev": int(rev)}
         raise GrafanaError("Failed to retrieve version from '%s'" % url)
 
-    def create_folder(self, title):
+    def create_folder(self, title, uid=None, parent_uid=None):
         url = "/api/folders"
-        folder = dict(title=title)
+        folder = dict(title=title, uid=uid, parentUid=parent_uid)
         response = self._send_request(
             url, data=folder, headers=self.headers, method="POST"
         )
         return response
 
-    def get_folder(self, title):
-        url = "/api/search?type=dash-folder&query={title}".format(title=quote(title))
+    def get_folder(self, title, uid=None, parent_uid=None):
+        url = "/api/folders%s" % ("?parentUid=%s" % parent_uid if parent_uid else "")
         response = self._send_request(url, headers=self.headers, method="GET")
-        for item in response:
-            if item.get("title") == to_text(title):
-                return item
+        if response is not None:
+            if uid:
+                folders = [item for item in response if item.get("uid") == uid]
+            else:
+                folders = [
+                    item for item in response if item.get("title") == to_text(title)
+                ]
+
+            if folders:
+                return folders[0]
```

Comment:
This would result in `folder[0]` only being returned for `else` and not for `if uid`. It should only be returned if folders are found, otherwise the following None should be returned.